packages/truffle-contract/lib/execute.js

```diff
@@ -110,9 +111,19 @@ var execute = {
       params.to = address;
       params = utils.merge(constructor.class_defaults, params);
 
-      return constructor
-        .detectNetwork()
-        .then(() => fn(...args).call(params, defaultBlock));
+      return new Promise(async (resolve, reject) => {
+        let result;
+        try {
+
+          await constructor.detectNetwork();
+          result = await fn(...args).call(params, defaultBlock);
+          result = reformat.callOutput.call(constructor, result, methodABI.outputs);
```

Comment:
Does this work? (won't it look for `methodABI.outputs.inputs`?)

Reply:
I *think* this is right. It's dereferencing the `outputs` keys of the object noted in the comment above . . .

Reply:
There are a couple changes upstream from here in `contract.js` that pass in a larger chunk of the ABI.
https://github.com/trufflesuite/truffle/pull/1138/files#diff-79802128c9314316abc8de8ceaa5bddcR51